Microsoft.NET.Build.Containers/Layer.cs

```diff
@@ -26,21 +27,26 @@ public static Layer FromFiles(IEnumerable<(string path, string containerPath)> f
     {
         long fileSize;
         Span<byte> hash = stackalloc byte[SHA256.HashSizeInBytes];
+        byte[] uncompressedHash;
 
         string tempTarballPath = ContentStore.GetTempFile();
         using (FileStream fs = File.Create(tempTarballPath))
         {
-            // using (GZipStream gz = new(fs, CompressionMode.Compress)) // TODO: https://github.com/dotnet/sdk-container-builds/issues/29
-            using (TarWriter writer = new(fs, TarEntryFormat.Gnu, leaveOpen: true))
+            using (HashDigestGZipStream gz = new(fs, leaveOpen: true))
             {
-                foreach (var item in fileList)
+                using (TarWriter writer = new(gz, TarEntryFormat.Gnu, leaveOpen: true))
```

Comment:
Question semi-unrelated to this change: @rainersigwald any reason why GNU format was used?

My assumption is that PAX was not used so that file size could be reduced, since it always created an extended attributes entry to precede each actual entry. That's fine, but just wanted to mention that PAX is the most flexible format, so if you encounter any issues, you can always try that format.